### ISSUE TITLE
Donación

### ISSUE THREAD

**Issue by rpasquini** [abcrecursos/abc-recursos-be#17]
La creación de una donación está avanzada, y a mi entender son solo pocas cosas las que hay que modificar. Debería incluir:

-  timestamp de cuando fue generada

- Los items donados. Son un array de ids de items (definidos en la coleccion supplies), cada uno con su cantidad asociada.

- Debe incluir el id de la persona que lo realiza. Esta persona tiene que crearse tambien como parte del proceso de donacion, si es que no existe ya. La persona que dona es un user que no pasó por el proceso de "registración en la plataforma" (validación de mail, contraseña etc).  El formulario contempla que deje sus datos de contacto: nombre, apellido, mail, telefono (Phone Schema), dirección y (luego de que complete la georeferenciacion) coordenadas geograficas.  Todo lo que tiene que ver con la direccion y coorderdanadas ya estan previsto en el Address Schema.

-Debe incluir al destinatario de la donación (esto no está en el front todavía). El algoritmo de matching le devolverá opciones de una serie de entidades de salud (healthcenter schema) u otras organizaciones (organizacion schema) para donar. La persona que dona elegirá un lugar a donde donar como parte del proceso. Por eso debe quedar registrada en la donación: el id del establecimiento de salud o el id de la organizacion. Las organizaciones a las que me refiero son las entidades que no son de salud que podrian entrar como receptores de donaciones.

- La donación debería **devolver** el codigo de seguimiento de la donación. Hablamos que este codigo seria una combinacion de la fecha y los ultimos cuatro digitos del Objectid, para que no sea tan feo de copiar como el ObjectId. Lo ideal seria que el codigo también se persista como un campo, para poder buscarla despues. Pero no me queda claro cuanto se puede complicar generar este campo.

**Comment by AlvaDamian** [comment]
Buenísimo :smile: .

Empiezo a trabajar con esto en la donación y, si sujen dudas o sugerencias, las dejo por acá para seguir con el hilo de la conversación.

Muchas gracias por tomarte el tiempo.

**Comment by AlvaDamian** [comment]
Ponelo como asignado para mí si es posible. Así el que lo vea sabe que se está trabajando en esto.

**Comment by AlvaDamian** [comment]
@rpasquini el modelo **Donations** debería ser el final?.

Estoy viendo que tiene un título, observaciones y prioridad. Creería que no serian necesarios desde el lado del donante. Por el momento dejaría estado actual de la donación, organización beneficiada y los ítems donados.

Actual
```typescript
const DonationSchema = new mongoose.Schema(
  {
    title: { type: String },
    observations: { type: String },
    priority: { type: Number, default: 5 },
    state: { type: String, enum: Object.keys(DonationStates) },
    healthCenter_id: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'HealthCenter',
    },
    items: [
      {
        supply_name: { type: String },
        supply_id: { type: mongoose.Schema.Types.ObjectId, ref: 'supply' },
        quantity: { type: Number },
      },
    ],
  },
  {
    timestamps: true,
  },
);
```

Nuevo:
```typescript
const DonationSchema = new mongoose.Schema(
  {
    state: { type: String, enum: Object.keys(DonationStates) },
    healthCenter_id: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'HealthCenter',
    },
    items: [
      {
        supply_name: { type: String },
        supply_id: { type: mongoose.Schema.Types.ObjectId, ref: 'supply' },
        quantity: { type: Number },
      },
    ],
  },
  {
    timestamps: true,
  },
);
```


Creo que:

- title: no aporta información a la donación y obliga al donante a pensar en algo o poner cualquier cosa.
- observations: ¿quién lo va a leer y que pondria el donante?
- priority: ¿quién define la prioridad de una donación?. ¿No debería tener prioridad la solicitud de donación y no la donación tal?

**Comment by rpasquini** [comment]
Sí, los campos que mencionas están de más. Para mí tampoco tienen lógica
alguna.
Faltaría lo que hay que agregar: el organization id cuando no es health
center, el usuario , etc
